src/edu/csus/ecs/pc2/clics/API202306/CLICSScoreboard.java

```diff
@@ -0,0 +1,129 @@
+
+        DefaultScoringAlgorithm scoringAlgorithm = new DefaultScoringAlgorithm();
+
+        Properties properties = ScoreboardUtilities.getScoringProperties(model);
+
+        ArrayList<Group> groupList = null;
+
+        if(group != null) {
+            groupList = new ArrayList<Group>();
+            groupList.add(group);
+        }
+        // legacy - standings are created as XML, and we convert that to JSON.
+        String xml = scoringAlgorithm.getStandings(model, null, division, groupList, properties, StaticLog.getLog());
+
+        ContestStandings contestStandings = ScoreboardUtilities.createContestStandings(xml);
+
+        // This is what we want to return:
+        //        {
+        //            "time": "2014-06-25T14:13:07.832+01",
+        //            "contest_time": "4:13:07.832",
+        //            "state": {
+        //              "started": "2014-06-25T10:00:00+01",
+        //              "ended": null,
+        //              "frozen": "2014-06-25T14:00:00+01",
+        //              "thawed": null,
+        //              "finalized": null,
+        //              "end_of_updates": null
+        //            },
+        //            "rows": [
+        //              {"rank":1,"team_id":"123","score":{"num_solved":3,"total_time":340},"problems":[
+        //                {"problem_id":"1","num_judged":3,"num_pending":1,"solved":false},
+        //                {"problem_id":"2","num_judged":1,"num_pending":0,"solved":true,"time":20},
+        //                {"problem_id":"3","num_judged":2,"num_pending":0,"solved":true,"time":55},
+        //                {"problem_id":"4","num_judged":0,"num_pending":0,"solved":false},
+        //                {"problem_id":"5","num_judged":3,"num_pending":0,"solved":true,"time":205}
+        //              ]}
+        //            ]
+        //          }
+        time = ZonedDateTime.now( ZoneOffset.UTC ).format( DateTimeFormatter.ISO_INSTANT);
+        contest_time = model.getContestTime().getElapsedTimeStr();
+        state = new CLICSContestState(model, null);
+
+        ArrayList<CLICSScoreboardRow>rowsArray = new ArrayList<CLICSScoreboardRow>();
+        HashMap<String, String> probEleToShort = new HashMap<String, String>();
+
+        // create a mapping of each problem's element ID to its shortname.
+        // we will use shortname as the problem id in the problem list for each team's solutions
+        for(Problem problem: model.getProblems()) {
+            probEleToShort.put(problem.getElementId().toString(), problem.getShortName());
+        }
+        // list of xml entries for each team's standing
+        List<TeamStanding> standings = contestStandings.getTeamStandings();
+        if(standings != null) {
+            for (TeamStanding teamStanding : contestStandings.getTeamStandings()) {
+              rowsArray.add(new CLICSScoreboardRow(probEleToShort, teamStanding));
+              rows = rowsArray.toArray(new CLICSScoreboardRow[0]);
```

Comment:
I don't understand why `rowsArray` is being converted (and assigned to `rows` inside the `for` loop.  It seems like this conversion should happen (once) after the `for` loop finishes...

Reply:
Yeah, neither do I.  I think I grabbed the essence of this code from somewhere else.  (I forget where).  And, it lost something in the translation to this module.   I can tell it was "stolen" from somewhere else because the indenting was wrong too.  (2 spaces instead of 4)   Fixed in next push.